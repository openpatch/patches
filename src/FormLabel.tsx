import { ReactNode } from "react";
import { css } from "@emotion/react";
import { Text } from "./Text";

export type FormLabelProps = {
  htmlFor: string;
  children?: ReactNode;
  required?: boolean;
};

export const FormLabel = ({
  htmlFor,
  children,
  required = false,
}: FormLabelProps) => {
  return (
    <Text
      mt="medium"
      mb="xsmall"
      fontWeight="medium"
      as="label"
      htmlFor={htmlFor}
      css={css`
        display: inline-block;
      `}
    >
      {children} {required && "*"}
    </Text>
  );
};
