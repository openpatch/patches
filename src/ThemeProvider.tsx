import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
