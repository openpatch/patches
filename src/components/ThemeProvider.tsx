import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
