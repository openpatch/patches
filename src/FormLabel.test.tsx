
import { render } from "@testing-library/react";
import { FormLabel, FormLabelProps } from "./FormLabel";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: FormLabelProps) =>
render(
    <ThemeProvider>
        <FormLabel {...props} />
    </ThemeProvider>
);

test("should render", () => {
   renderWithTheme();
});

test("should match snapshot", () => {
    const { container } = renderWithTheme();
    expect(container).toMatchSnapshot();
});