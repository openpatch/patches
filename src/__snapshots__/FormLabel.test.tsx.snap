// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  padding: 0px;
  margin: 0px;
  margin-top: 24px;
  margin-bottom: 8px;
  font-weight: 500;
  display: inline-block;
}

<div>
  <div
    id="patches-root"
  >
    <label
      class="emotion-0"
      font-weight="medium"
    >
       
    </label>
  </div>
</div>
`;
